src/py2app/build_app.py

```diff
@@ -1741,6 +1741,12 @@ def files_in_dir(
                 mm.excludes.append(runtime)
             else:
                 mm.mm.run_file(runtime)
+
+            dylib_dir = os.path.dirname(runtime)
+            for fn in os.listdir(dylib_dir):
+                if fn.endswith(".dylib"):
+                    mm.mm.run_file(os.path.join(dylib_dir, fn))
```

Comment:
Found a bug! Some dylibs are nested in further directories, and so aren't copied. The may also have to be copied as-specified? 

Therefore, it may be better to copy the whole environment — it seems like best practice too, keeping a minimal VENV.

What HASN'T been answered, is why did this work for so long, and then suddenly stop working, and for (maybe) apple silicon only? I hope a maintainer who knows the code picks this up. I spent enough time, so I'm done here.